src/DistributedArrays.jl

```diff
@@ -880,13 +880,37 @@ for f in (:+, :-, :div, :mod, :rem, :&, :|, :$)
             B = samedist(A, B)
             map_localparts($f, A, B)
         end
+        function ($f){T}(A::DArray{T}, B::Array{T})
+            size(A) == size(B) || throw(DimensionMismatch("first array has size $(size(A)) but second array has $(size(B))"))
+            return DArray(A) do I
+                ($f)(localpart(A), B[I...])
```

Comment:
It depends on how it's done. If we use the `construct_darray` function I think we'd have the same problem. I think we'd have to slice `B` before the closure and that is a bit messy because the indices are calculated inside the `DArray` constructor. Another issue with the solution right now is that I think we end up serializing the closure with `B` `nprocs(A)` times instead of only once.